hardware/arduino/avr/cores/arduino/WString.cpp

```diff
@@ -193,7 +193,7 @@ String & String::copy(const __FlashStringHelper *pstr, unsigned int length)
 void String::move(String &rhs)
 {
 	if (buffer) {
-		if (capacity >= rhs.len) {
+		if (rhs && capacity >= rhs.len) {
 			strcpy(buffer, rhs.buffer);
 			len = rhs.len;
 			rhs.len = 0;
```

Comment:
> the only downside is that the buffer pointer for this string changes (but nobody should rely on it anyway)?

there is the `c_str()` operator and there are PR that pushes for using it outside the string class also for writing :-)

> Maybe it's to avoid memory fragmentation?

I guess that's the reason, `move` is called from the c++11 move-constructor, so it's probably better to keep the original (rvalue) allocation whenever possible to avoid making "holes".